### PR TITLE
1117 bugfix casa admin download csv report

### PR DESCRIPTION
**Pull request by erik-trantt**
### What github issue is this PR for, if any?
Resolves the first 2 bugs mentioned in #1117 

### What changed, and why?
Refactored scope `contact_type_groups` to fix unhandled bug for 
- unhandled situation when user selects the prompt option (with value as am empty string) in Contact Type Group filter
  - `where("contact_types.contact_type_group_id in (?)", contact_type_group_ids)`: `contact_type_group_id` is expecting a `bigint` type, but is given empty string `""`
- duplicate records when `case_contact` involves many `contact_types` of the same `contact_type_group`

```rb
# app/models/case_contact.rb  

  scope :contact_type_groups, ->(contact_type_group_ids = nil) {
    # to handle case when passing ids == [''] && ids == nil
    if contact_type_group_ids&.join&.length&.positive?
      joins(contact_types: :contact_type_group)
        .where(contact_type_groups: { id: contact_type_group_ids })
        .group(:id)
    end
  }
```

In test file `app/models/case_contact.rb`, move other test examples out of "occured at range filter" text example. So that, it makes more sense when looking from high level view. See screenshot:

image

### How will this affect user permissions?
N/A

### How is this tested? (please write tests!) 💖💪
Refactored and added tests for Contact Type Group filter to cover for edge cases when 
- user selects prompt option => contact_type_group_ids = [""] 
- user selects prompt option and other group types => contact_type_group_ids = ["", "1", "2"] 
- user selects nothing in the Contact Type Group filter => contact_type_group_ids = nil

Test cases locates at
  - `app/models/case_contact.rb`

### Screenshots please :)
image

**Report outcome when select 1 group type: CASA & with / without prompt option**
image

**Report outcome when select prompt option / select nothing**
image

### CODE REVIEW

File: spec/models/case_contact_report_spec.rb [hunk 1]
```diff
@@ -78,166 +78,204 @@
...
+      end
+    end
 
-          report = CaseContactReport.new({want_driving_reimbursement: true})
-          contacts = report.case_contacts
-          expect(contacts.length).to eq(1)
-        end
+    describe "wanting driving reimbursement functionality" do
+      it "returns only contacts that want reimbursement" do
+        # want_driving_reimbursement
```
Comment (compwron) [inline spec/models/case_contact_report_spec.rb:143]: This comment is not needed

**Review by compwron — APPROVED** [review]
This is a great and very explanatory PR! Thanks for the bug fixes and great tests :) ✨💪

File: spec/models/case_contact_report_spec.rb (hunk 1 above)
Comment (erik-trantt) [reply spec/models/case_contact_report_spec.rb:143]: > In test file `app/models/case_contact.rb`, move other test examples out of "occured at range filter" text example. So that, it makes more sense when looking from high level view. See screenshot:

Good find. That bit is unnecessary.
I think it was there before, cos' I just worked on the "contact type group filter functionality" and indented the rest.

Comment (erik-trantt) [reply spec/models/case_contact_report_spec.rb:143]: I didnt realize that reindenting code would create this situation. When I used `git diff` on VSCode, it just showed the code I added and ignore reindentation. 😅
